### DISCUSSION TITLE
Upgrade old modules

### DISCUSSION THREAD

**Discussion by moderator10000078** [Zrips/CMI#7501] [Enhancement]
unfortunately, there are modules in CMI that no one uses, in my opinion they should be updated

The Combat module does not have the necessary functionality, for example, killing when exiting the server or when kicking (2 separate types of killing), cooldown on items during PVP (stands on many servers), and in comparison with combatlogx or Deluxe combat, it looks weak (there are a lot of settings in them, for example, creating fake walls when trying to log into the worldguard region in which PVP is disabled

Sit module - there is only /sit in this module, although there are many other poses in the game, for example, lying down, lying on your stomach, etc., many just use Gsit, because it has all this, and even more, for example, you can sit on other players

GUI - does not have a full-fledged configuration, in a good way I would like to see a full-fledged GUI creation right in minecraft without taking in configs

TAB and Nameplates - the problem with TAB is that it is only on the TAB itself, but there is no scoreboard, nicknames have a problem in the absence of multi-line nicknames, so everyone uses TAB from Neznamy

Almost all of the above are modules necessary for many servers, but unfortunately they are quite weak in CMI, I would like them to be updated

**Comment by MrCrazys** [comment]
> unfortunately, there are modules in CMI that no one uses, in my opinion they should be updated
> 
> The Combat module does not have the necessary functionality, for example, killing when exiting the server or when kicking (2 separate types of killing), cooldown on items during PVP (stands on many servers), and in comparison with combatlogx or Deluxe combat, it looks weak (there are a lot of settings in them, for example, creating fake walls when trying to log into the worldguard region in which PVP is disabled
> 
> Sit module - there is only /sit in this module, although there are many other poses in the game, for example, lying down, lying on your stomach, etc., many just use Gsit, because it has all this, and even more, for example, you can sit on other players
> 
> GUI - does not have a full-fledged configuration, in a good way I would like to see a full-fledged GUI creation right in minecraft without taking in configs
> 
> TAB and Nameplates - the problem with TAB is that it is only on the TAB itself, but there is no scoreboard, nicknames have a problem in the absence of multi-line nicknames, so everyone uses TAB from Neznamy
> 
> Almost all of the above are modules necessary for many servers, but unfortunately they are quite weak in CMI, I would like them to be updated

I don't think it's fair to say that "there are modules in CMI that no one uses.", because you're effectively saying that for every server owner that uses CMI.

CMI isn't designed to have super advanced features, like the "creating walls" feature. It can't replace every single plugin. CMI also isn't a permission plugin, nor is it a GUI builder or anticheat.

Again, CMI isn't designed to replace all plugins. There are some features that are just out of the scope of the plugin. By the way, you can sit on other players by using `/cmi ride`.

There is already customization for the GUIs of CMI. You effectively have complete control over the home, kit and warp GUIs. CMI isn't designed to be a complete, all-in-one GUI plugin. If you need lots of customization, use a proper GUI plugin such as CommandPanels.

TAB is a very advanced tab plugin, and it doesn't have as many other features as CMI. Scoreboards just seem out of the scope of CMI. It's best just to use a scoreboard plugin such as FeatherBoard. And multi-line nicknames just seem unnecessary.

I don't see a lot of problems with the current modules of CMI. Sure, they could use some improvements, but I don't think they need any major changes or rewrites.
